wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -295,25 +312,162 @@ class PresentationActivity : FragmentActivity() {
+                                    authKeyLockedCredential = it
+                                }, consentData = consentPromptData.value)
+                            }
+                        }
+
+                        // user is viewing biometrics flow
+                        PresentationFlow.BiometricsPrompt -> {
+                            when (flowReturnState.value) {
+                                // the return state from current or previous flow is "locked auth key"
+                                PresentationFlow.ReturnState.LOCKED_AUTHENTICATION_KEY -> {
+                                    // show the biometrics prompt
+                                    onAuthenticationKeyLocked(
+                                        authKeyLockedCredential!!,
+                                        flowReturnState
+                                    )
+                                }
+
+                                // we have a valid key unlock data and auth key to use for retrying finishing processing request
+                                PresentationFlow.ReturnState.SUCCESS -> {
+                                    finishProcessingRequest(
+                                        keyUnlockData = biometricsPromptData!!.keyUnlockData,
+                                        authKey = biometricsPromptData!!.authKey,
+                                        onAuthKeyLocked = {
+                                            // if it's still locked we'll need to re-prompt (shouldn't happen)
+                                            flowReturnState.value =
+                                                PresentationFlow.ReturnState.LOCKED_AUTHENTICATION_KEY
+                                            // for this credential
+                                            authKeyLockedCredential = it
+                                        },
+                                        onFinishedProcessingRequest = {
+                                            // track the processed bytes to pass to the global lambda for submitting the request
+                                            processedByteArray = it
+                                            // change set flow to passphrase
+                                            presentationFlow.value =
+                                                PresentationFlow.PassphrasePrompt
+                                            flowReturnState.value =
+                                                PresentationFlow.ReturnState.INIT
+                                        },
+                                        consentData = consentPromptData.value
+                                    )
+                                }
+
+                                else -> {}
+                            }
+                        }
+
+                        // user is on the Passphrase flow
+                        PresentationFlow.PassphrasePrompt -> {
+
+                        }
```

Comment:
Hmm, I thought we talked about changing this it's not callback-based but the presentation is just one linear suspend function?